iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/GpuIcebergPartitioner.scala

```diff
@@ -208,4 +208,77 @@ object GpuIcebergPartitioner {
           }).toArray
     }
   }
+
+  private def addRowIdxToTable(table: Table): Table = {
```

Comment:
Please update after this PR is merged: https://github.com/NVIDIA/spark-rapids/pull/13688